### PR TITLE
[GCS]Optimize gcs client nodes get function

### PR DESCRIPTION
**Pull request by ffbin**
## Why are these changes needed?
We filter dead nodes from gcs_client->Nodes(). `Get()` returns optional none if the node is already dead.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by rkooo567** [comment]
cc @stephanie-wang Before I approve, I wonder if there's any use case for having the ability to get node information that is already dead?

**Comment by ffbin** [comment]
> I wonder if there's any use case for having the ability to get node information that is already dead?

The dead node will be dropped by evict, so it is of little significance to obtain the dead node.

**Comment by stephanie-wang** [comment]
What is this change for? Sometimes it's useful to be able to tell if a node has died.

**Comment by ffbin** [comment]
@stephanie-wang https://github.com/ray-project/ray/blob/35c10c94bc81fe7a705578cf51140031be1d5b88/src/ray/core_worker/core_worker.cc#L1179 Here we need to check the node status, because the node state maybe dead. And now we will evict dead node, if you want to determine whether a node is dead, you may not get it.

**Comment by rkooo567** [comment]
Screen Shot 2020-10-20 at 10 33 59 PM

For more details, this is the image @ffbin sent to me.

**Comment by rkooo567** [comment]
If there's any use case that we'll need to get dead nodes, we can probably just have a default arg like "filter_dead_node", which is true by default.

**Comment by ffbin** [comment]
> If there's any use case that we'll need to get dead nodes, we can probably just have a default arg like "filter_dead_node", which is true by default.

Good idea, thanks.

**Comment by ffbin** [comment]
Hi @rkooo567 This PR can be reviewed, thanks.

**Review by rkooo567 — APPROVED** [review]

**Pull request merged**
